pallets/asset-manager/src/lib.rs

```diff
@@ -374,17 +498,173 @@ pub mod pallet {
+			T::LocalAssetModifierOrigin::ensure_origin(origin)?;
+
+			// Get the deposit amount
+			let deposit = T::LocalAssetDeposit::get();
+
+			// Verify we can reserve
+			T::Currency::can_reserve(&creator, deposit)
+				.then(|| true)
+				.ok_or(Error::<T>::NotSufficientDeposit)?;
+
+			// Read Local Asset Counter
+			let mut local_asset_counter = LocalAssetCounter::<T>::get();
+
+			// Create the assetId with LocalAssetIdCreator
+			let asset_id =
+				T::LocalAssetIdCreator::create_asset_id_from_metadata(local_asset_counter);
+
+			// Increment the counter
+			local_asset_counter = local_asset_counter
+				.checked_add(1)
+				.ok_or(Error::<T>::LocalAssetLimitReached)?;
 
-			Self::deposit_event(Event::AssetRemoved {
+			// Create local asset
+			T::AssetRegistrar::create_local_asset(
+				asset_id,
+				creator.clone(),
+				min_balance,
+				is_sufficient,
+				owner.clone(),
+			)
+			.map_err(|_| Error::<T>::ErrorCreatingAsset)?;
+
+			// Reserve the deposit, we verified we can do this
+			T::Currency::reserve(&creator, deposit)?;
+
+			// Update assetInfo
+			LocalAssetDeposit::<T>::insert(
+				asset_id,
+				AssetInfo {
+					creator: creator.clone(),
+					deposit,
+				},
+			);
+
+			// Update local asset counter
+			LocalAssetCounter::<T>::put(local_asset_counter);
+
+			Self::deposit_event(Event::LocalAssetRegistered {
+				asset_id,
+				creator,
+				owner,
+			});
+			Ok(())
+		}
+
+		/// Destroy a given foreign assetId
+		/// The weight in this case is the one returned by the trait
+		/// plus the db writes and reads from removing all the associated
+		/// data
+		#[pallet::weight({
+			let dispatch_info_weight = T::AssetRegistrar::destroy_asset_dispatch_info_weight(
+				*asset_id, *destroy_asset_witness
+			);
+			T::WeightInfo::remove_existing_asset_type(*num_assets_weight_hint)
+			.saturating_add(dispatch_info_weight)
+		})]
+		pub fn destroy_foreign_asset(
+			origin: OriginFor<T>,
+			asset_id: T::AssetId,
+			destroy_asset_witness: T::AssetDestroyWitness,
+			num_assets_weight_hint: u32,
+		) -> DispatchResult {
+			T::ForeignAssetModifierOrigin::ensure_origin(origin)?;
+
+			T::AssetRegistrar::destroy_foreign_asset(asset_id, destroy_asset_witness)
+				.map_err(|_| Error::<T>::ErrorDestroyingAsset)?;
+
+			// Grab supported assets
+			let mut supported_assets = SupportedFeePaymentAssets::<T>::get();
+
+			ensure!(
+				num_assets_weight_hint >= (supported_assets.len() as u32),
+				Error::<T>::TooLowNumAssetsWeightHint
+			);
+
+			let asset_type =
+				AssetIdType::<T>::get(&asset_id).ok_or(Error::<T>::AssetDoesNotExist)?;
+
+			// Remove from AssetIdType
+			AssetIdType::<T>::remove(&asset_id);
+			// Remove from AssetTypeId
+			AssetTypeId::<T>::remove(&asset_type);
+			// Remove previous asset type units per second
+			AssetTypeUnitsPerSecond::<T>::remove(&asset_type);
+
+			// Only if the old asset is supported we need to remove it
+			if let Ok(index) = supported_assets.binary_search(&asset_type) {
```

Comment:
More of a note-to-self: make sure this vec is sorted in all cases

Reply:
I think we always insert them retrieving their potential index with binary_search. So I think this should be enough to guarantee they are sorted